### DISCUSSION TITLE
Github Actions CI/CD with SnowDDL

### DISCUSSION THREAD

**Discussion by PradeepReddy-epi** [littleK0i/SnowDDL#38] [Q&A]
Hi,
    I'm automating SnowDDL with GH actions. Plan when PR raised ,then review plan output and trigger apply workflow on merge which applies only safe queries. Any solutions to ensure that merge PR on latest plan output (i.e only apply reviewed plan) , may be just like in terraform, we can apply only the plan which is generated on plan step. I know snowddl is stateless.
    
Thanks

**Comment by littleK0i** [comment]
You may use SnowDDL to generate "Suggested SQL" only with `plan` mode, but apply it via different means after review. For example, you may use basic Python script with standard connector to apply reviewed SQL as plain-text.

Please note: state of objects in Snowflake account might change between start and finish of the review process, so some queries may still fail.
